multi-arch-builders/coreos-builder-splunk.bu

```diff
@@ -0,0 +1,103 @@
+#
+variant: fcos
+version: 1.4.0
+passwd:
+  users:
+    - name: splunk
+      groups:
+        - systemd-journal
+        - wheel
+storage:
+  directories:
+    - path: /home/splunk/.config
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd/user
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/systemd/user/default.target.wants
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/containers
+      user:
+        name: splunk
+      group:
+        name: splunk
+    - path: /home/splunk/.config/containers/systemd
+      user:
+        name: splunk
+      group:
+        name: splunk
+  files:
+    - path: /var/lib/systemd/linger/splunk
+      mode: 0644
+    - path: /home/splunk/.config/containers/systemd/splunk.container
+      mode: 0644
+      user:
+        name: splunk
+      group:
+        name: splunk
+      contents:
+        inline: |
+          [Unit]
+          Description=Container for splunk
+          After=build-splunk.service
+          [Container]
+          ContainerName=rhcos-multiarch-splunk
+          Image=localhost/splunkforwarder-rhcos:latest
+          RunInit=true
+          User=root
+          Volume=/var/log/journal:/var/log/journal:ro
+          Volume=/run/log/journal:/run/log/journal:ro
+          Volume=/etc/machine-id:/etc/machine-id:ro
+          Volume=/var/home/core:/var/log/journald:ro
+          Network=host
+          PodmanArgs=--ipc=host --group-add keep-groups --hostname=${SPLUNK_HOSTNAME}
+          [Install]
+          WantedBy=default.target
+    - path: /home/splunk/.config/systemd/user/build-splunk.service
+      mode: 0644
+      user:
+        name: splunk
+      group:
+        name: splunk
+      contents:
+        inline: |
+          [Unit]
+          Description=Build splunk container
+          [Service]
+          # Give time for the build to complete
+          TimeoutStartSec=10m
+          Type=oneshot
+          RemainAfterExit=yes
+          # Figure out soon if we can get the Red Hat Certs onto the hosts
+          Environment=GIT_SSL_NO_VERIFY=1
+          ExecStartPre=nm-online --timeout=30
+          # use `--arch=x86_64` because the RPM doesnt exist for aarch64. ppc64le and s390x RPMs dont work for journald input.
+          ExecStart=-podman build --pull-always --cache-ttl=480h --arch=x86_64 -t localhost/splunkforwarder:latest https://gitlab.corp.redhat.com/paas/cicd.git#main:itpaas-util-images/splunkforwarder
+          ExecStart=-podman build --build-arg project=${SPLUNK_PROJECT} --cache-ttl=480h --from localhost/splunkforwarder:latest -t localhost/splunkforwarder-rhcos https://gitlab.corp.redhat.com/paas/docker-paas.git#master:splunk
```

Comment:
Is there no way to build the container and just pull it from the registry here?
Building the container each time on reboot sounds failure prone and unnecessary.

Also, if using pre-built container images, it would be possible to poll for updates regulary, instead of relying on reboots for updates. Not sure how often that would happen, but we might want to have those run unattended, as that is most of the reason we are using fcos for the pipeline.

Reply:
@dustymabe and I talked about this. If we push it to a registry, we most likely need it locked down somewhere. That would mean configuring extra credentials on the builders when we install. Building the container each time also reduces the burden of maintaining a separate container build workflow.

Reply:
Is it also possible to hide the internal gitlab address?

Reply:
If you already talked through the implications of building it and decided it is the better option i am fine with that.

What about the automated updates? I guess right now it would happen each time fcos is updated (rebooted) right? Is that frequent enough?